### PR TITLE
Yrax protocols part1

### PR DESCRIPTION
**Pull request by Vgoloshivskiy**
#### Summary
Content "Slightly Expands Hub-01 and Yrax factions"

#### Purpose of change
Yrax don't have a whole lot of content associated with them. I added a bit, plan on adding more content

#### Describe the solution
It adds to Yrax something more than just optional fight

#### Describe alternatives you've considered

image

image
I have no motivation to rewrite entire conversation "("

#### Testing

Worked without crashes but Hub reacted to robots only if i was holding them in inventory idk is it fixable or not

#### Additional context

Next part i will try myself at map editing and creating Gold Pyramid that reacts to electricity (being shot by laser / chain lightning from zed or emp grenade) or maybe being connected to generator

### CODE REVIEW

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 1]
```diff
@@ -232,5 +232,235 @@
...
+      "effect": [
+        { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" }
+      ],
+      "sentinel": "yrax_trade"
+    }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_QUADRAPHRACT",
+    "type": "talk_topic",
+    "dynamic_line": "I'm impressed you managed to drag it inside.  We gonna need to fix lift after this one.",
```
Comment (emixa-d) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:414]: `lift` -> `the lift`

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 2]
```diff
@@ -232,5 +232,235 @@
...
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_DELTA"
+      },
+      { "text": "That would be all.", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" }
+    ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIFACET",
+    "type": "talk_topic",
+    "dynamic_line": "I belive we already have similar model in better condition but is nice to have something for reference.",
```
Comment (emixa-d) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:359]: `belive` -> `believe`

File: data/json/items/corpses/inactive_bots.json [hunk 3]
```diff
@@ -754,6 +754,61 @@
...
+    "type": "TOOL",
+    "symbol": "I",
+    "name": { "str": "inactive golden monolith" },
+    "description": "A large golden monolith.  Carefully carved asymmetric edges betray its artificial origin.",
+    "volume": "930 L",
+    "//": "A lot of gold in this bad boy.",
+    "weight": "8282 kg",
+    "price": "50 USD",
+    "price_postapoc": "50 USD",
+    "melee_damage": { "bash": 1000 },
```
Comment (emixa-d) [inline data/json/items/corpses/inactive_bots.json:785]: Moving this massive thing would be so slow I doubt you could damage anyone with it, even if they stand still (except maybe damaging their feet). (And unless it falls from a weight, but IIUC that is it's own thing, separate from me `melee_damage`.) So I think the `melee_damage` should be lower.

Comment (Vgoloshivskiy) [reply data/json/items/corpses/inactive_bots.json:785]: I'm pretty sure i copied it from broken bot but Ill look into it.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 4]
```diff
@@ -232,5 +232,235 @@
...
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIFACET"
+      },
+      {
+        "text": "[Put in broken triakias.]",
+        "condition": { "u_has_items": { "item": "broken_yrax_triakis", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "broken_yrax_triakis", "count": 1, "popup": true },
+          { "math": [ "u_broken_yrax_triakis_sold", "++" ] },
+          { "u_spend_cash": -1000 }
```
Comment (emixa-d) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:269]: Any reason why sometimes credit is given directly whereas other times they give a physical token of their currency instead? Seems inconsistent.

Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:269]: credit is given for broken/ small robots ; real tokens are quite expensive

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 5]
```diff
@@ -232,5 +232,235 @@
...
+      "effect": [
+        { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" }
+      ],
+      "sentinel": "yrax_trade"
+    }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BOT_GOLDEN_MONOLITH",
+    "type": "talk_topic",
+    "dynamic_line": "I assume you were trying to figure out limits of hm12.  I guess they are right here.",
```
Comment (emixa-d) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:407]: Probably `the hm12`,, but I think this works too in this case.

File: data/json/vehicleparts/stowed_parts.json [hunk 6]
```diff
@@ -472,5 +472,73 @@
...
+  {
+    "type": "vehicle_part",
+    "id": "stowed_broken_golden_monolith",
+    "name": { "str": "tied-down broken golden monolith" },
+    "categories": [ "other" ],
+    "color": "light_blue",
+    "damage_modifier": 5,
+    "durability": 20,
+    "//": "This is used to simulate a rope being untied/broken and falling off",
+    "description": "More than a ton of gold tied to roof for easier transportation.  Probably the only way to move this without breaking your back.",
```
Comment (emixa-d) [inline data/json/vehicleparts/stowed_parts.json:519]: In the vehicle construction menu, does an appropriate lifting tool requirement appear?

Comment (Vgoloshivskiy) [reply data/json/vehicleparts/stowed_parts.json:519]: of course without boom crane you can't lift this thing

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 5 above)
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:407]: not sure about that either

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 4 above)
Comment (emixa-d) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:269]: Do I understand correctly that -1000 = -10$ (which is in turn <1 Hub coin)? (Where $ is this nebulous post-apocalypse "dollar".) If so, that seems reasonable since the Hub does not issue "Hub cents".

File: data/json/items/nether.json [hunk 7]
```diff
@@ -12,7 +12,7 @@
     "volume": "250 ml",
     "weight": "10 g",
     "symbol": "@",
-    "description": "This defies easy description, it feels solid but the surface is yielding.  Looking at it you have a strange desire to call it a door.",
+    "description": "This defies easy description, it feels solid but the surface is yielding.  Looking at it you have a strange desire to call it a shelter.",
```
Comment (emixa-d) [inline data/json/items/nether.json:15]: I don't have an opinion on this change, but this currently seems unrelated to the title of this PR and I don't see an obvious reason for this change. Are you perhaps planning on adding more commits, some of which expect this item to be a shelter of sorts?

Comment (Vgoloshivskiy) [reply data/json/items/nether.json:15]: is is shelter already (it creates 5 minutes long safe space for you) when i was starting this PR i changed it as i was orienting more on Yrax conversation (in alternatives you can read) but it was quickly shut down so i changed what I was doing with this PR but it remained and i belive that to be better description then the same description as rift.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 5 above)
Comment (emixa-d) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:407]: I think:  "figure out limits of hm12" = "figure out the limits of the class of rifles named hm12", "figure out limits of the hm12" = either "figure out the limits of the platonic hm12" or "figure out the limits of your hm12". In this case, while technically different messages, there isn't any particular difference.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 8]
```diff
@@ -232,5 +232,185 @@
...
+        "condition": { "u_has_items": { "item": "broken_yrax_triakis", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "broken_yrax_triakis", "count": 1, "popup": true },
+          { "math": [ "u_broken_yrax_triakis_sold", "++" ] },
+          { "u_spend_cash": -1000 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIAKIS"
+      },
+      {
+        "text": "[Roll in broken monolith.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:274]:
```suggestion
        "text": "[Deliver a broken monolith.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 9]
```diff
@@ -232,5 +232,185 @@
...
+        "condition": { "u_has_items": { "item": "broken_yrax_quadraphract", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "broken_yrax_quadraphract", "count": 1, "popup": true },
+          { "math": [ "u_broken_yrax_quadraphract_sold", "++" ] },
+          { "u_spawn_item": "RobofacCoin", "count": 25 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_QUADRAPHRACT"
+      },
+      {
+        "text": "[Put in deactivated trifacet.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:314]:
```suggestion
        "text": "[Deliver a deactivated trifacet.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 10]
```diff
@@ -232,5 +232,185 @@
...
+        "condition": { "u_has_items": { "item": "bot_yrax_trifacet", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "bot_yrax_trifacet", "count": 1, "popup": true },
+          { "math": [ "u_bot_yrax_trifacet_sold", "++" ] },
+          { "u_spawn_item": "RobofacCoin", "count": 2 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIFACET"
+      },
+      {
+        "text": "[Put in deactivated triakis.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:324]:
```suggestion
        "text": "[Deliver a deactivated triakis.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 11]
```diff
@@ -232,5 +232,185 @@
...
+        "condition": { "u_has_items": { "item": "bot_yrax_triakis", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "bot_yrax_triakis", "count": 1, "popup": true },
+          { "math": [ "u_bot_yrax_triakis_sold", "++" ] },
+          { "u_spawn_item": "RobofacCoin", "count": 1 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIAKIS"
+      },
+      {
+        "text": "[Roll in deactivated monolith.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:334]:
```suggestion
        "text": "[Deliver a deactivated monolith.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 12]
```diff
@@ -232,5 +232,185 @@
...
+        "condition": { "u_has_items": { "item": "bot_golden_monolith", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "bot_golden_monolith", "count": 1, "popup": true },
+          { "math": [ "u_bot_golden_monolith_sold", "++" ] },
+          { "u_spawn_item": "RobofacCoin", "count": 16 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BOT_GOLDEN_MONOLITH"
+      },
+      {
+        "text": "[Put in deactivated delta.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:344]:
```suggestion
        "text": "[Deliver a deactivated delta.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 13]
```diff
@@ -232,5 +232,185 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIFACET",
+    "type": "talk_topic",
+    "dynamic_line": "I believe we already have similar model in better condition but is nice to have something for reference.",
+    "speaker_effect": { "effect": [ { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" } ], "sentinel": "yrax_trade" }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIAKIS",
+    "type": "talk_topic",
+    "dynamic_line": "We can look at this but it appears to miss quite a lot to it.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:365]:
```suggestion
    "dynamic_line": "We can probably gain some insight from studying this unit, but  it appears seriously damaged",
```

**Comment by John-Candlebury** [comment]
I'm not particularily sold on the player being able to disable the rest of the drones beyond the trifacet. The trifacet is only deactivable because the US government managed to recover and study at least one before the world eneded.

**Comment by Vgoloshivskiy** [comment]
> I'm not particularily sold on the player being able to disable the rest of the drones beyond the trifacet. The trifacet is only deactivable because the US government managed to recover and study at least one before the world eneded.

> I honestly didn't see any check in code that checked did you read trifacet handling procedures or not (could have missed it)
I partially agree i want to add couple changes later that will allow you to deactivate rest if you bring to Hub-01 broken and let them work their magic for a bit (but not right now as it is already quite a lot of code in this)

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 14]
```diff
@@ -232,5 +232,235 @@
...
+      },
+      {
+        "text": "[Drag in broken quadraphract.]",
+        "condition": { "u_has_items": { "item": "broken_yrax_quadraphract", "count": 1 } },
+        "effect": [
+          { "u_consume_item": "broken_yrax_quadraphract", "count": 1, "popup": true },
+          { "math": [ "u_broken_yrax_quadraphract_sold", "++" ] },
+          { "u_spawn_item": "RobofacCoin", "count": 25 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_QUADRAPHRACT"
```
[the remark this answers is not in the document]
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:311]: It is more complex i will agree with you on that but i added it as vehicle part so you don't have to drag it all the way.
And camp with mercenaries again I never did maps and don't think i can learn as this week i have ton of work irl

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 13 above)
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json]: You have talent for writing

**Comment by Vgoloshivskiy** [comment]
> I'm not particularily sold on the player being able to disable the rest of the drones beyond the trifacet. The trifacet is only deactivable because the US government managed to recover and study at least one before the world eneded.

Just added checks might need more testing than i did, but should looks nice

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 4 above)
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:274]: Yes

File: data/json/items/corpses/inactive_bots.json (hunk 3 above)
Comment (Vgoloshivskiy) [reply data/json/items/corpses/inactive_bots.json]: Fixed

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 2 above)
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json]: corrected

File: gfx/GiantDays/tile_config.json [hunk 15]
```diff
@@ -8493,7 +8493,7 @@
             "cookbook_sushi",
             "holybook_granth",
             "holybook_scientology",
-            "robofac_yrax_deactivation_manual"
```
Comment (John-Candlebury) [inline gfx/GiantDays/tile_config.json:8496]: You should not expand the tile_config.json files manually. 

I'm pretty sure these will just get undone or cause conflicts in the next automated update for the tilesets.

File: src/activity_actor.cpp [hunk 16]
```diff
@@ -243,6 +243,20 @@ static const move_mode_id move_mode_prone( "prone" );
...
+
+static const itype_id
+itype_robofac_yrax_trifacet_deactivation_manual( "robofac_yrax_trifacet_deactivation_manual" );
+static const itype_id
+itype_robofac_yrax_triakis_deactivation_manual( "robofac_yrax_triakis_deactivation_manual" );
+static const itype_id
+itype_robofac_yrax_golden_monolith_deactivation_manual( "robofac_yrax_golden_monolith_deactivation_manual" );
+static const itype_id
+itype_robofac_yrax_delta_deactivation_manual( "robofac_yrax_delta_deactivation_manual" );
+
```
Comment (John-Candlebury) [inline src/activity_actor.cpp:259]: You need to fix the failing checks in this file.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 17]
```diff
@@ -232,5 +232,263 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIFACET",
+    "type": "talk_topic",
+    "dynamic_line": "I believe we already have similar model already but is nice to have more, it will allow us be more active in our research.",
+    "speaker_effect": { "effect": [ { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" } ], "sentinel": "yrax_trade" }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIAKIS",
+    "type": "talk_topic",
+    "dynamic_line": "Oh what an interesting piece R&D will have a field day with it.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:479]:
```suggestion
    "dynamic_line": "Oh what an interesting piece.  R&D will have a field day with it.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 18]
```diff
@@ -232,5 +232,263 @@
...
+      "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIFACET",
+      "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_TRIAKIS",
+      "TALK_ROBOFAC_INTERCOM_SELL_BOT_GOLDEN_MONOLITH",
+      "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_DELTA"
+    ],
+    "type": "talk_topic",
+    "dynamic_line": "Beautiful; please put them into loading area.  We'll issue a reward depending on what you managed to scavenge.",
+    "responses": [
+      {
+        "text": "[Put in broken trifacet.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:254]:
```suggestion
        "text": "[Deliver a broken trifacet.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 19]
```diff
@@ -232,5 +232,263 @@
...
+        },
+        "effect": [
+          { "u_consume_item": "broken_yrax_delta", "count": 1, "popup": true },
+          { "math": [ "u_broken_yrax_delta_sold", "++" ] },
+          { "u_spend_cash": -200 }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_BROKEN_YRAX_DELTA"
+      },
+      {
+        "text": "[Put in broken triakias.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:309]: This appears to be a repeat of the option above.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 20]
```diff
@@ -232,5 +232,263 @@
...
+        "effect": [
+          { "u_consume_item": "broken_golden_monolith", "count": 1, "popup": true },
+          { "math": [ "u_timer_hub_rnd_u_waiting_on_research", "=", "time('now')" ] },
+          { "u_add_var": "dialogue_hub_rnd_u_gave_broken_golden_monolith", "value": "yes" },
+          { "u_add_var": "dialogue_hub_rnd_u_project_ongoing", "value": "yes" }
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_ACCEPT_BROKEN_YRAX"
+      },
+      {
+        "text": "[Put in broken delta.]",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:341]:
```suggestion
        "text": "[Deliver a broken delta.]",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 21]
```diff
@@ -232,5 +232,263 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_GOLDEN_MONOLITH",
+    "type": "talk_topic",
+    "dynamic_line": "That is quite a lot of broken metal, let's hope R&D can at least salvage materials.",
+    "speaker_effect": { "effect": [ { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" } ], "sentinel": "yrax_trade" }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_DELTA",
+    "type": "talk_topic",
+    "dynamic_line": "You don't need to pick up all glossy scrap that you find.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:455]:
```suggestion
    "dynamic_line": "We don't need all of your glossy scrap.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 22]
```diff
@@ -232,5 +232,263 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIAKIS",
+    "type": "talk_topic",
+    "dynamic_line": "We can probably gain some insight from studying this unit, but  it appears seriously damaged",
+    "speaker_effect": { "effect": [ { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" } ], "sentinel": "yrax_trade" }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_GOLDEN_MONOLITH",
+    "type": "talk_topic",
+    "dynamic_line": "That is quite a lot of broken metal, let's hope R&D can at least salvage materials.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:449]:
```suggestion
    "dynamic_line": "That is quite a lot of broken metal, let's hope R&D can at least salvage some materials.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 23]
```diff
@@ -232,5 +232,263 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIFACET",
+    "type": "talk_topic",
+    "dynamic_line": "I believe we already have similar model in better condition but it is nice to have something for reference.",
+    "speaker_effect": { "effect": [ { "npc_add_var": "dialogue_intercom_accepted_yrax_robot", "value": "yes" } ], "sentinel": "yrax_trade" }
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIAKIS",
+    "type": "talk_topic",
+    "dynamic_line": "We can probably gain some insight from studying this unit, but  it appears seriously damaged",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:443]:
```suggestion
    "dynamic_line": "We can probably gain some insight from studying this unit, but  it appears seriously damaged.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 24]
```diff
@@ -232,5 +232,263 @@
...
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_YRAX_TOYS_ONGOIN_RESEARCH",
+    "type": "talk_topic",
+    "dynamic_line": "We don't have any free hands to process your new findings.",
+    "responses": [ { "text": "I see, I'll bother you later then.", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" } ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_TRIFACET",
+    "type": "talk_topic",
+    "dynamic_line": "I believe we already have similar model in better condition but it is nice to have something for reference.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:437]:
```suggestion
    "dynamic_line": "We already have a similar model in better condition but this one is useful as a reference.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 25]
```diff
@@ -232,5 +232,263 @@
...
+    "dynamic_line": "Come back later, we are not sure how soon we can finish studying this remains",
+    "responses": [
+      { "text": "What else do you have?", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" },
+      { "text": "See you later then.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_SELL_YRAX_TOYS_ONGOIN_RESEARCH",
+    "type": "talk_topic",
+    "dynamic_line": "We don't have any free hands to process your new findings.",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:431]:
```suggestion
    "dynamic_line": "We don't have the capacity to process your findings at the moment.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json [hunk 26]
```diff
@@ -232,5 +232,263 @@
...
+        ],
+        "topic": "TALK_ROBOFAC_INTERCOM_SELL_BOT_YRAX_DELTA"
+      },
+      { "text": "That would be all.", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" }
+    ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_ACCEPT_BROKEN_YRAX",
+    "type": "talk_topic",
+    "dynamic_line": "Come back later, we are not sure how soon we can finish studying this remains",
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:422]:
```suggestion
    "dynamic_line": "Come back later, we are not sure how soon we can finish studying these remains",
```

**Review by John-Candlebury — CHANGES_REQUESTED** [review]
Adding more deactivation manuals feels like a copout. I think you should just make it impossible for the player to disable the Yrax bots other than the Trifacet.

You can keep the option to deliver inactive drones to the Hub, but the player has to find them in inactive forms. You can add more dormant Yrax that work similar to the Sphenocoronae in future PRs.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json (hunk 19 above)
Comment (Vgoloshivskiy) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json:309]: It isn't top is for selling, this one is for research

**Comment by Vgoloshivskiy** [comment]
> Adding more deactivation manuals feels like a copout. I think you should just make it impossible for the player to disable the Yrax bots other than the Trifacet.
> 
> You can keep the option to deliver inactive drones to the Hub, but the player has to find them in inactive forms. You can add more dormant Yrax that work similar to the Sphenocoronae in future PRs.

I did't want for it to feel like copout, I wanted something like this to begin with.
It is not hard to remove this functionality by just removing revert to item from yrax bots json.
I might add spawn locations with inactive Yrax but i wanted there to be possibility to disable Yrax and other robots if you know how to.
Monolith is a dormant Sphenocoronae but I can't really deliver it as it is still a monster and not an item that i can put in car.

**Review by John-Candlebury — APPROVED** [review]

**Comment by MikasaTanikawa** [comment]
@Vgoloshivskiy please add migration for new manual id and at https://github.com/Vgoloshivskiy/Cataclysm-DDA/blob/bf05b72514cbee078b6a37c86791c585c5795549/data/json/npcs/robofac/robofac_intercom/robofac_intercom_trades.json#L291 should be "TALK_ROBOFAC_INTERCOM_SELL_BROKEN_YRAX_DELTA"
